### PR TITLE
Add API method to rename devices

### PR DESCRIPTION
**Pull request by thomseddon**
Not sure on how you'd like this method to be presented? Let me know if you'd prefer the new hostname to be in the body (I just wanted some kind of differentiation).

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: html/includes/api_functions.inc.php [hunk 1]
```diff
@@ -1262,6 +1262,41 @@ function update_device()
...
+        $message = 'Missing new hostname';
+    } elseif ($new_device) {
+        $message = 'Device failed to rename, new hostname already exists.';
+    } else {
+        if (renamehost($device_id, $new_hostname, 'api') == '') {
+            $status = 'ok';
+            $code = 200;
+            $message = 'Device has been renamed';
+        } else {
+            $message = 'Device failed to be reanmed';
```
Comment (murrant) [inline html/includes/api_functions.inc.php]: small typo here `reanmed` -> `renamed`

**Comment by laf** [comment]
Thanks @thomseddon.

The API code went through a little refactor the other week so you'll need to update the code a bit to match (I can do that if you haven't got time?)

**Comment by thomseddon** [comment]
Fixed the typo, thanks for spotting that.

I've also added `check_is_admin();` to match `update_device`?

@laf what branch should I rebase from? I've just checked and it seems up to date with master on here, just let me know what I'm missing, can get it fixed today :)

**Comment by laf** [comment]
> @laf what branch should I rebase from? I've just checked and it seems up to date with master on here, just let me know what I'm missing, can get it fixed today :)

@thomseddon sorry for the delay, busy week as you know.

So sending an error back can be done like: https://github.com/librenms/librenms/blob/master/html/includes/api_functions.inc.php#L255

Sending the end response can be done like: https://github.com/librenms/librenms/blob/master/html/includes/api_functions.inc.php#L263

Happy to update if you don't have time.

**Comment by thomseddon** [comment]
@laf ah sorry, I see it now - my mistake!

I've switched to the new functions to form the responses (much nicer :+1:) and tided up a couple of other things (erroneous new line, unnecessary full stop in response)

Happy to make any more amends :)

**Comment by laf** [comment]
Thanks for doing this @thomseddon :)

**Pull request merged**